NOMIS/nomis.go

```diff
@@ -301,3 +307,25 @@ func downloadFile() {
 
 	fmt.Printf("Downloaded a file %s with size %d", fileName, size)
 }
+
+func checkSubString (res CenStructure, index0, indx int)  string {
+	var strArray []string
+	var result string
+	value :=strings.Contains(res.Structure.Keyfamilies.Keyfamily[index0].Annotations.Annotation[indx].Text.(string), "http")
+	if value{
+		str1:=strings.Split(res.Structure.Keyfamilies.Keyfamily[index0].Annotations.Annotation[indx].Text.(string), "http")
+		var valueCheck = regexp.MustCompile("html|.aspx")
+		strMatch:=valueCheck.MatchString(str1[1])
```

Comment:
If you were to "fail fast"  on this check you can avoid having to further nest the logic.

```
...
if !strMatch {
    return result
}

if strings.Contains( str1[1],"html") {
    strArray = strings.SplitAfter(str1[1], "html")
 } else if strings.Contains( str1[1],"aspx") {
    strArray = strings.SplitAfter(str1[1], "aspx")
 }
 result=fmt.Sprintf("%s%s%s%s%s", str1[0],"(http", strArray[0], ")",strArray[1])
 ...
}
```